nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1851,7 +1851,7 @@ protected boolean pack() {
             configurationGenerator.setProperty(PARAM_UPDATECENTER_DISABLED, "true");
             List<String> startCommand = new ArrayList<>();
             startCommand.add(getJavaExecutable().getPath());
-            startCommand.addAll(getJavaOptsProperty(Function.<String>identity()));
+            startCommand.addAll(getJavaOptsProperty(opt -> StrSubstitutor.replace(opt, configurationGenerator.getUserConfig())));
```

Comment:
Don't have any warning about duplicated code, so it might be related to another issue, no? And if it is required, it must be part of the rewrite of this class... 'cause it is still more than 3 thousands lines long...